addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/ZoneControlXML.java

```diff
@@ -0,0 +1,209 @@
+        } else {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Volume><Mute>Off</Mute></Volume>"));
+        }
+        update();
+    }
+
+    @Override
+    public void setInput(String name) throws IOException, ReceivedMessageParseException {
+        // ToDo: See the fixme in the class description
+        comReference.get().send(XMLUtils.wrZone(zone, "<Input><Input_Sel>" + name + "</Input_Sel></Input>"));
+        update();
+    }
+
+    @Override
+    public void setSurroundProgram(String name) throws IOException, ReceivedMessageParseException {
+        if (name.toLowerCase().equals("straight")) {
+            comReference.get().send(XMLUtils.wrZone(zone,
+                    "<Surround><Program_Sel><Current><Straight>On</Straight></Current></Program_Sel></Surround>"));
+        } else {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Surround><Program_Sel><Current><Sound_Program>" + name
+                    + "</Sound_Program></Current></Program_Sel></Surround>"));
+        }
+
+        update();
+    }
+
+    @Override
+    public void update() throws IOException, ReceivedMessageParseException {
+        if (observer == null) {
+            return;
+        }
+
+        AbstractConnection com = comReference.get();
+        String response = com.sendReceive(XMLUtils.wrZone(zone, "<Basic_Status>GetParam</Basic_Status>"));
+        Document doc = XMLUtils.xml(response);
+        if (doc.getFirstChild() == null) {
+            throw new ReceivedMessageParseException("<Basic_Status>GetParam failed: " + response);
+        }
+        Node basicStatus = XMLUtils.getNode(doc.getFirstChild(), zone + "/Basic_Status");
+
+        String value;
+
+        ZoneControlState state = new ZoneControlState();
+
+        value = XMLUtils.getNodeContentOrDefault(basicStatus, "Power_Control/Power", "");
+        state.power = "On".equalsIgnoreCase(value);
+
+        value = XMLUtils.getNodeContentOrDefault(basicStatus, "Input/Input_Sel", "");
+        // ToDo: See the fixme in the class description
+        state.inputID = XMLUtils.convertNameToID(value);
+
+        if (StringUtils.isBlank(state.inputID)) {
+            throw new ReceivedMessageParseException(
+                    "Expected inputID. Failed to read Input/Input_Sel_Item_Info/Src_Name");
+        }
+
+        // Some receivers may use Src_Name instead?
+        value = XMLUtils.getNodeContentOrDefault(basicStatus, "Input/Input_Sel_Item_Info/Title", "");
+        state.inputName = value;
+
+        value = XMLUtils.getNodeContentOrDefault(basicStatus, "Surround/Program_Sel/Current/Sound_Program", "");
+        state.surroundProgram = value;
+
+        Node node = XMLUtils.getNode(basicStatus, "Volume/Lvl/Val");
+        value = node != null ? node.getTextContent() : String.valueOf(YamahaReceiverBindingConstants.VOLUME_MIN);
+        state.volume = Float.parseFloat(value) * .1f; // in DB
```

Comment:
Double is default so I see no reason to force it be a float*

\* I was trained that float should never be used anymore because the little memory saving is not worth the potential inaccuracy